### PR TITLE
Replace moment with dayjs

### PR DESCRIPTION
**Pull request by qqlexa**
Rebased from qqlexa:replace-axios-with-fetch-api

### CODE REVIEW

File: src/common/utils/generateScheduleMatrix.ts [hunk 1]
```diff
@@ -19,13 +19,14 @@ export const generateScheduleMatrix = <T extends Pair,>(weekSchedule: Schedule<T
...
-  const currentDay = moment().day();
+  const currentDay = dayjs().day();
 
   weekSchedule.forEach((schedule) => {
     const yIndex = DAYS.findIndex((item) => item === schedule.day);
 
     schedule.pairs.forEach((pair) => {
-      const xIndex = TIME_POINTS.findIndex((item) => item === pair.time);
+      const normalizedPairTime = createDateTime(pair.time);
+      const xIndex = TIME_POINTS.findIndex((item) => item === normalizedPairTime.format('HH:mm'));
```
Comment (niravzi) [inline src/common/utils/generateScheduleMatrix.ts:29]: You could format `normalizedPairTime` before using it inside `.findIndex` function.

File: src/api/index.ts [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
-});
+const defaultHeaders = {
+  "Content-Type": "application/json",
+};
+
+interface FetchOptions extends RequestInit {
+  headers?: Record<string, string>;
+}
+
+const fetchWrapper = async (post_url: string, options: FetchOptions = {}) => {
```
Comment (niravzi) [inline src/api/index.ts:11]: - Follow naming conventions: `post_url` -> `postUrl`
- Please choose better naming for this parameter (something like `requestUrl`)

File: src/api/index.ts [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+
+  const requestOptions: RequestInit = {
+    ...restOptions,
+    headers: {
+      ...defaultHeaders,
+      ...headers,
+    },
+  };
+
+  try {
```
Comment (niravzi) [inline src/api/index.ts:22]: This `try-catch` block looks useless.

**Review by niravzi — CHANGES_REQUESTED** [review]

**Comment by a-gubskiy** [comment]
@qqlexa any update?

**Comment by qqlexa** [comment]
> @qqlexa any update?

Updates only on weekends 😉

File: src/api/fullList.ts [hunk 4]
```diff
@@ -3,16 +3,16 @@ import { Lecturer } from '../models/Lecturer';
 import { PagedResponse } from '../models/PagedResponse';
 import Http from "./index";
 
-export const getAllLecturers = (): Promise<PagedResponse<Lecturer[]>> => {
-  return Http.get("/schedule/lecturer/list").then(({ data }) => data);
+export const getAllLecturers = async (): Promise<PagedResponse<Lecturer[]>> => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/api/fullList.ts]: Updated

File: src/api/getCurrentDateValues.ts [hunk 5]
```diff
@@ -2,6 +2,6 @@ import { CurrentTime } from '../models/CurrentTime';
 import { PagedResponse } from '../models/PagedResponse';
 import Http from './index';
 
-export const getCurrentDateValues = (): Promise<PagedResponse<CurrentTime>> => {
-  return Http.get('/time/current').then(res => res.data);
-};
\ No newline at end of file
+export const getCurrentDateValues = async (): Promise<PagedResponse<CurrentTime>> => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/api/getCurrentDateValues.ts]: Updated

File: src/api/schedule.ts [hunk 6]
```diff
@@ -4,17 +4,14 @@ import { PagedResponse } from '../models/PagedResponse';
 import { StudentLesson } from '../models/StudentLesson';
 import Http from './index';
 
-export const getScheduleByLecturer = (lecturerId: string): Promise<PagedResponse<LecturerLesson[]>> => {
-  return Http.get('/schedule/lecturer?lecturerId=' + lecturerId)
-    .then(res => res.data);
+export const getScheduleByLecturer = async (lecturerId: string): Promise<PagedResponse<LecturerLesson[]>> => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/api/schedule.ts]: Updated

File: src/api/schedule.ts [hunk 7]
```diff
@@ -4,17 +4,14 @@ import { PagedResponse } from '../models/PagedResponse';
...
+  return await Http.get('/schedule/lecturer?lecturerId=' + lecturerId);
 }
 
-export const getScheduleByGroup = (groupId: string): Promise<PagedResponse<StudentLesson[]>> => {
-  return Http.get('/schedule/lessons?groupId=' + groupId)
-    .then(res => res.data);
+export const getScheduleByGroup = async (groupId: string): Promise<PagedResponse<StudentLesson[]>> => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/api/schedule.ts]: Updated

File: src/api/schedule.ts [hunk 8]
```diff
@@ -4,17 +4,14 @@ import { PagedResponse } from '../models/PagedResponse';
...
+  return await Http.get('/schedule/lessons?groupId=' + groupId);
 }
 
-export const getExamsByGroup = (groupName: string): Promise<PagedResponse<Exam[]>> => {
-  return Http.get('/exams/group?groupId=' + groupName)
-    .then(res => res.data);
-}
\ No newline at end of file
+export const getExamsByGroup = async (groupName: string): Promise<PagedResponse<Exam[]>> => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/api/schedule.ts]: Updated

File: src/common/utils/getActiveTimePoint.ts [hunk 9]
```diff
@@ -1,15 +1,31 @@
-import moment from 'moment';
+import dayjs from 'dayjs';
+import isBetween from 'dayjs/plugin/isBetween';
+import duration from 'dayjs/plugin/duration';
 import { TIME_POINTS } from '../../common/constants/scheduleParams';
 
+dayjs.extend(isBetween);
+dayjs.extend(duration);
+
+export const createDateTime = (timeString: string) => {
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/common/utils/getActiveTimePoint.ts]: Updated

File: src/common/utils/getActiveTimePoint.ts [hunk 10]
```diff
@@ -1,15 +1,31 @@
...
+  const [hours, minutes] = timeString.split(':').map(Number);
+
+  const today = dayjs().startOf('day');
+
+  return today
+      .hour(hours)
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/common/utils/getActiveTimePoint.ts]: Updated

File: src/components/examSchedule/examSchedule.tsx [hunk 11]
```diff
@@ -1,11 +1,17 @@
...
 import teacherIcon from '../../assets/icons/teacher.svg';
 import locationIcon from '../../assets/icons/location.svg';
 import clock from '../../assets/icons/clock.svg';
 import { Flex, Pictogram } from '../../common/styles/styles';
 import React from 'react';
 import { Exam } from '../../models/Exam';
-import moment from 'moment';
+
+// Extend Day.js with the required plugins
+dayjs.extend(localizedFormat);
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/components/examSchedule/examSchedule.tsx]: Done

File: src/components/examSchedule/examSchedule.tsx [hunk 12]
```diff
@@ -35,7 +41,7 @@ const ExamSchedule = ({ exam }: Props) => {
             </Teacher>
             <Teacher>
               <Pictogram src={clock} alt="time"/>
-              {time.format('h:mm A')}
+              {time}
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/components/examSchedule/examSchedule.tsx]: Sure, done
Question: which group can I use to check this behavior?

File: src/components/timeDivider/timeDivider.tsx [hunk 13]
```diff
@@ -9,7 +8,7 @@ import {
 const TimeDivider = ({ value }: { value: string }) => (
   <Container>
     <InnerContainer>
-      <TimeCell>{moment(value, 'H:mm').format('HH:mm')}</TimeCell>
+      <TimeCell>{ value }</TimeCell>
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/components/timeDivider/timeDivider.tsx:11]: Done

File: src/common/utils/generateScheduleMatrix.ts (hunk 1 above)
Comment (qqlexa) [reply src/common/utils/generateScheduleMatrix.ts:29]: Yes, good idea. Done

File: src/api/index.ts (hunk 2 above)
Comment (qqlexa) [reply src/api/index.ts:11]: Thank you.
Renamed to requestUrl

File: src/api/index.ts (hunk 3 above)
Comment (qqlexa) [reply src/api/index.ts]: Okay, removed

File: src/components/examSchedule/examSchedule.tsx [hunk 14]
```diff
@@ -18,9 +24,9 @@ const ExamSchedule = ({ exam }: Props) => {
     room,
   } = exam;
 
-  const date = moment(exam.date);
-  const time = moment(exam.date).locale('en-US');
-  const daysLeft = moment.duration(date.diff(moment())).days();
+  const date = dayjs(exam.date);
```
[the remark this answers is not in the document]
Comment (qqlexa) [reply src/components/examSchedule/examSchedule.tsx:21]: The duplicated one was removed

**Comment by qqlexa** [comment]
Comments were resolved

**Review by niravzi — APPROVED** [review]

**Pull request merged**